ansible_collections/arista/avd/roles/eos_designs/doc/fabric-variables.md

```diff
@@ -61,6 +61,39 @@ underlay_ospf_area: < ospf_area | default -> 0.0.0.0 >
 underlay_ospf_max_lsa: < lsa | default -> 12000 >
 underlay_ospf_bfd_enable: < true | false | default -> false >
 
+ospf_router_id_source_interface:
```

Comment:
Some of these are overlapping existing keys. Why implement them again?

Since most of this is directly copied from `eos_cli_config_gen` without any abstraction, I am not a big fan of adding all this to eos_designs as global vars.
Also it is not clear if these variables will affect both underlay ospf process as well as any overlay ospf processes. We try to use `underlay_` prefix to make it clear where the variables are leveraged.
I see two ways forward:
1. Use `custom_structured_configuration_router_ospf` (my preference and suggestion)
2. Create a root-level variable `underlay_ospf_settings` and insert all these variables there. It could be implemented fairly simple by just taking eos_cli_config_gen variables 1:1 rendering them under the correct ospf process.